New Package: fcitx-chewing-0.2.3_1 & libchewing-0.5.1_1

CorporalCrawdad opened the pull request:
Add ZhuYin support through fcitx with libchewing

maxice8 commented on srcpkgs/fcitx-chewing/template:
```diff
@@ -0,0 +1,19 @@
+version=0.2.3
+revision=1
+build_style=cmake
+#configure_args=""
```
remove commented variables

maxice8 commented on srcpkgs/fcitx-chewing/template:
```diff
@@ -0,0 +1,19 @@
+depends="fcitx libchewing"
+short_desc="Fcitx wrapper for Chewing IM engine"
+maintainer="Robert Stancil <robert.stancil@mavs.uta.edu>"
+license="GPL-2"
```
use SPDX identifier

maxice8 commented on srcpkgs/fcitx-chewing/template:
```diff
@@ -0,0 +1,19 @@
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="pkg-config"
+makedepends="fcitx-devel libchewing-devel"
+depends="fcitx libchewing"
```
this will be found via the shlib tracking system, no need to add it.

maxice8 commented on srcpkgs/libchewing-devel/template:
```diff
@@ -0,0 +1,34 @@
+pkgname=libchewing
+version=0.5.1
+revision=1
+#wrksrc=
```
remove commented variables

maxice8 commented on srcpkgs/libchewing-devel/template:
```diff
@@ -0,0 +1,34 @@
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="autoconf automake libtool clang gcc"
+makedepends="sqlite-devel"
+depends="sqlite"
```
remove

maxice8 commented on srcpkgs/libchewing-devel/template:
```diff
@@ -0,0 +1,34 @@
+		vmove usr/lib/pkgconfig
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/share
```
Please make sure there is no files in /usr/share that belong in the main package and not in the -devel package, be aware that in updates that might change so it is recomended to vmove the correct paths under /usr/share

maxice8 commented on srcpkgs/libchewing-devel/template:
```diff
@@ -0,0 +1,34 @@
+hostmakedepends="autoconf automake libtool clang gcc"
+makedepends="sqlite-devel"
+depends="sqlite"
+short_desc="An intelligent phonetic (Zhuyin/Bopomofo) input method"
```
Do not use Article 'An'

maxice8 commented on srcpkgs/libchewing/template:
```diff
@@ -0,0 +1,35 @@
+pkgname=libchewing
+version=0.5.1
+revision=1
+#wrksrc=
```
Remove commented variable

maxice8 commented on srcpkgs/libchewing/template:
```diff
@@ -0,0 +1,35 @@
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="autoconf automake libtool clang gcc"
+makedepends="sqlite-devel"
+depends="sqlite"
```
Remove

maxice8 replied:
ping

CorporalCrawdad replied:
All the changes have been made in the most recent commit, are you saying this shouldn't depend on sqlite?

maxice8 commented on srcpkgs/libchewing/template:
```diff
@@ -0,0 +1,35 @@
+#make_install_args=""
+#conf_files=""
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="autoconf automake libtool clang gcc"
```
gcc is part of the base-system, remove it.

CorporalCrawdad replied:
auto* is part of base-devel, does it need to go too?

maxice8 commented on srcpkgs/libchewing/template:
```diff
@@ -0,0 +1,35 @@
+#make_install_args=""
+#conf_files=""
+#make_dirs="/var/log/dir 0755 root root"
+hostmakedepends="autoconf automake libtool clang gcc"
```
is clang really required ?

maxice8 commented on srcpkgs/libchewing/template:
```diff
@@ -0,0 +1,35 @@
+hostmakedepends="autoconf automake libtool clang gcc"
+makedepends="sqlite-devel"
+depends="sqlite"
+short_desc="An intelligent phonetic (Zhuyin/Bopomofo) input method"
```
Do not use article 'An'

maxice8 commented on srcpkgs/libchewing/template:
```diff
@@ -0,0 +1,35 @@
+homepage="http://chewing.im/"
+distfiles="https://github.com/chewing/libchewing/releases/download/v${version}/${pkgname}-${version}.tar.bz2"
+checksum="9708c63415fa6034435c0f38100e7d30d0e1bac927f67bec6dfeb3fef016172b"
+shlib_provides="libchewing.so.3"
```
I don't think this is necessary.

CorporalCrawdad replied:
I think this is an artifact of misunderstanding the comon/shlibs file. Removed

maxice8 replied:
This is only necessary when you need to provide shlib tracking for internal libraries (those that live under /usr/lib/pkgname/lib.so instead of /usr/lib/lib.so) See `mutter` as an example

maxice8 commented on srcpkgs/libchewing/template:
```diff
@@ -0,0 +1,35 @@
+		vmove usr/lib/pkgconfig
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.so"
+		vmove usr/share
```
Same warnings apply here

maxice8 commented:
libchewing-devel should be a symlink from libchweing to libchewing-devel

maxice8 commented on srcpkgs/libchewing-devel/template:
```diff
@@ -0,0 +1,34 @@
+# Template file for 'libchewing'
```
Turn this file into a symlink, see the files inside srcpkg/

maxice8 commented:
On Sat, Mar 23, 2019 at 11:40:14AM -0700, CorporalCrawdad wrote:
> CorporalCrawdad commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,35 @@
> +# Template file for 'libchewing'
> +pkgname=libchewing
> +version=0.5.1
> +revision=1
> +#wrksrc=
> +#create_wrksrc=yes
> +#only_for_archs="i686 x86_64"
> +build_style=cmake
> +#configure_args=""
> +#make_build_args=""
> +#make_install_args=""
> +#conf_files=""
> +#make_dirs="/var/log/dir 0755 root root"
> +hostmakedepends="autoconf automake libtool clang gcc"
> 
> auto* is part of base-devel, does it need to go too?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10188#discussion_r268404787

No, i used the wrong package. You should be looking at base-chroot. My bad.

Johnnynator commented:
libchewing is not cross compiling, set `nocross=" /builddir/libchewing-0.5.1/build/src/tools/init_database: cannot execute binary file"`

maxice8 commented:
On Sun, Mar 24, 2019 at 09:25:04AM -0700, CorporalCrawdad wrote:
> CorporalCrawdad commented on this pull request.
> 
> 
> 
> > +# Template file for 'libchewing'
> +pkgname=libchewing
> +version=0.5.1
> +revision=1
> +#wrksrc=
> +#create_wrksrc=yes
> +#only_for_archs="i686 x86_64"
> +build_style=cmake
> +#configure_args=""
> +#make_build_args=""
> +#make_install_args=""
> +#conf_files=""
> +#make_dirs="/var/log/dir 0755 root root"
> +hostmakedepends="autoconf automake libtool clang gcc"
> +makedepends="sqlite-devel"
> +depends="sqlite"
> 
> All the changes have been made in the most recent commit, are you saying this shouldn't depend on sqlite? 
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10188#discussion_r268441419

sqlite-devel is in makedepends it most problably means that it will link against the sqlite library which will be automatically found and added to the depends= via out shlib tracking system.

CorporalCrawdad commented:
> On Sun, Mar 24, 2019 at 09:25:04AM -0700, CorporalCrawdad wrote: CorporalCrawdad commented on this pull request. > +# Template file for 'libchewing' +pkgname=libchewing +version=0.5.1 +revision=1 +#wrksrc= +#create_wrksrc=yes +#only_for_archs="i686 x86_64" +build_style=cmake +#configure_args="" +#make_build_args="" +#make_install_args="" +#conf_files="" +#make_dirs="/var/log/dir 0755 root root" +hostmakedepends="autoconf automake libtool clang gcc" +makedepends="sqlite-devel" +depends="sqlite" All the changes have been made in the most recent commit, are you saying this shouldn't depend on sqlite? -- You are receiving this because you commented. Reply to this email directly or view it on GitHub: [#10188 (comment)](https://github.com/void-linux/void-packages/pull/10188#discussion_r268441419)
> sqlite-devel is in makedepends it most problably means that it will link against the sqlite library which will be automatically found and added to the depends= via out shlib tracking system.
It sure does. Removed. This shlib tracking sure is neato.

maxice8 commented:
merged with fixes to the commit history and miscellaneous ones.

ixearth commented:
anyone tested it and work for you?

I have fcitx, fcitx-chewing, fcitx-configtool, libfcitx-{gtk, gtk3, qt, qt5} installed and with dbus running. also set the environment variable. tried on xfce4 (musl, glibc) and gnome (glibc).
both does not works.
after install the package, chewing appear in the list on fcitx-configtool or I can add it. but I can't switch to the input method.
and later it will disappear from the list.

I even tried on arch linux, installed xorg, xfce4, related fcitx packages (need to install fcitx-im the group which basically are libfcitx-*). and it works there.

CorporalCrawdad commented:
@ixearth This works for me, however I can say that many applications don't handle IM support well. Make sure GTK_IM_MODULE and XMODIFIERS is set, and make sure that the application you are testing with supports fcitx.

ixearth commented:
someone I met on telegram tested fcitx-chewing on void-glibc and he have the same situation as me. chewing appears in input method list but after try to ctrl+shift to switch to it, it can't be switched to and will be disappear in input method list.
and fcitx-mozc works for him, I also tried both fcitx-anthy and fcitx-mozc works for me.
Pinyin the fcitx built-in module also works for me.
so it's most likely a problem directly related to fcitx-chewing rather than fcitx.

I already have environment variable set.
```
export GTK_IM_MODULE=fcitx
export XMODIFIERS=@im=fcitx
export QT_IM_MODULE=fcitx
export QT4_IM_MODULE=fcitx
export QT5_IM_MODULE=fcitx
```
and I have `libfcitx{,-gtk,-gtk3,-qt,-qt5}` installed. dbus service is enabled and running.
I have no idea why it does not works.
@CorporalCrawdad can you tell me more details of your set-up? (glibc or musl, which DM/WM, any extra packages you installed etc?)

CorporalCrawdad commented:
I'm on glibc, using xinit and dwm, so nothing fancy
I don't have any other related packages installed, though I can confirm the method switching is pretty poor. Manually switching in fcitx-configtool works, but I figured this was just a symptom of not using a full-fledged DE.

ixearth commented:
that's strange. I just tried fcitx-chewing on dwm on void-glibc (x86_64) it still does not works.
after start dwm by using startx, I manually run `fcitx` in st, and then I press ctrl+space to active fcitx. ctrl+alt can't switch to chewing.
fcitx-configtool seems can be used only for adding/removing input method, I think only shortcuts and fcitx on tray could be use for switch input method.
but I didn't manage to add a tray for dwm yet.

Francesco149 commented:
manually switching from the fcitx-configtool tray makes chewing disappear from the list and switches to other available input methods instead and the only way to make it re-appear is restarting fcitx. so yeah, it's broken for me as well

ixearth commented:
voidlinux-glibc (x86_64)
installed fcitx packages:
```
$ xbps-query -l | grep fcitx
ii fcitx-4.2.9.6_3                           Flexible Context-aware Input Tool with eXtension
ii fcitx-chewing-0.2.3_1                     Fcitx wrapper for Chewing IM engine
ii fcitx-configtool-0.4.10_1                 A GTK-based configuration tool for fcitx
ii libfcitx-4.2.9.6_3                        Flexible Context-aware Input Tool with eXtension - shared libraries
ii libfcitx-gtk-4.2.9.6_3                    Flexible Context-aware Input Tool with eXtension - GTK2 IM module
ii libfcitx-gtk3-4.2.9.6_3                   Flexible Context-aware Input Tool with eXtension - GTK3 IM module
ii libfcitx-qt-4.2.9.6_3                     Flexible Context-aware Input Tool with eXtension - Qt4 IM module
ii libfcitx-qt5-1.2.3_1                      Flexible Context-aware Input Tool with eXtension - Qt5 IM module
```

I tried fcitx-chewing on dwm 6.1 with systray, when I select chewing from fcitx on systray, fcitx output `Chewing storage path /home/username/.config/fcitx/chewing/` in terminal emulator and chewing disappear from input list.
```
$ fcitx
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-xkbdbus.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-imselector.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-qw.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-autoeng.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-xim.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-spell.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-kimpanel-ui.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-ipcportal.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-vk.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-clipboard.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-ipc.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-classic-ui.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-pinyin.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-chttrans.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-remote-module.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-punc.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-notificationitem.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-keyboard.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-fullwidth-char.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-table.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-unicode.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-dbus.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-freedesktop-notify.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-quickphrase.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-pinyin-enhance.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-x11.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-xkb.conf
(INFO-6766 addon.c:151) Load Addon Config File:fcitx-chewing.conf
(ERROR-6766 ime.c:432) fcitx-keyboard-in-kan-kagapa already exists
(ERROR-6766 ime.c:432) fcitx-keyboard-in-tel-kagapa already exists
(ERROR-6766 ime.c:432) fcitx-keyboard-cm-mmuock already exists
(INFO-6766 eim.c:105) Chewing storage path /home/username/.config/fcitx/chewing/
(WARN-6766 x11selection.c:309) Selection is too long.
```

```
$ fcitx-diagnose
# System Info:
1.  `uname -a`:

        Linux void 4.19.34_1 #1 SMP PREEMPT Sun Apr 7 08:49:36 UTC 2019 x86_64 GNU/Linux

2.  `lsb_release -a`:

        LSB Version:	1.0
        Distributor ID:	VoidLinux
        Description:	Void Linux
        Release:	rolling
        Codename:	void

3.  `lsb_release -d`:

        Description:	Void Linux

4.  `/etc/lsb-release`:

    `/etc/lsb-release` not found.

5.  `/etc/os-release`:

        NAME="void"
        ID="void"
        DISTRIB_ID="void"
        PRETTY_NAME="void"

6.  Desktop Environment:

    Cannot determine desktop environment.

7.  Bash Version:

        BASH_VERSION='5.0.3(1)-release'

# Environment:
1.  DISPLAY:

        DISPLAY=':0'

2.  Keyboard Layout:

    1.  `setxkbmap`:

            xkb_keymap {
            	xkb_keycodes  { include "evdev+aliases(qwerty)"	};
            	xkb_types     { include "complete"	};
            	xkb_compat    { include "complete"	};
            	xkb_symbols   { include "pc+us(dvp)+inet(evdev)"	};
            	xkb_geometry  { include "pc(pc104)"	};
            };

    2.  `xprop`:

            _XKB_RULES_NAMES(STRING) = "evdev", "pc104", "us", "dvp", ""

3.  Locale:

    1.  All locale:

            C
            POSIX
            en_US.utf8
            zh_TW.utf8

    2.  Current locale:

            LANG=en_US.UTF-8
            LC_CTYPE="en_US.UTF-8"
            LC_NUMERIC="en_US.UTF-8"
            LC_TIME="en_US.UTF-8"
            LC_COLLATE=C
            LC_MONETARY="en_US.UTF-8"
            LC_MESSAGES=
            LC_PAPER="en_US.UTF-8"
            LC_NAME="en_US.UTF-8"
            LC_ADDRESS="en_US.UTF-8"
            LC_TELEPHONE="en_US.UTF-8"
            LC_MEASUREMENT="en_US.UTF-8"
            LC_IDENTIFICATION="en_US.UTF-8"
            LC_ALL=

4.  Directories:

    1.  Home:

            /home/username

    2.  `${XDG_CONFIG_HOME}`:

        Environment variable `XDG_CONFIG_HOME` is not set.

        Current value of `XDG_CONFIG_HOME` is `~/.config` (`/home/username/.config`).

    3.  Fcitx Settings Directory:

        Current fcitx settings directory is `~/.config/fcitx` (`/home/username/.config/fcitx`).

5.  Current user:

    The script is run as username (1000).

# Fcitx State:
1.  executable:

    Found fcitx at `/bin/fcitx`.

2.  version:

    Fcitx version: `4.2.9.5`

3.  process:

    Found 2 fcitx processes:

         6766 fcitx
         9335 fcitx-dbus-watc

4.  `fcitx-remote`:

    `fcitx-remote` works properly.

5.  DBus interface:

    Using `dbus-send` to check dbus.

    Owner of DBus name `org.fcitx.Fcitx` is `:1.11`.

    PID of DBus name `org.fcitx.Fcitx` owner is `6766`.

# Fcitx Configure UI:
1.  Config Tool Wrapper:

    Found fcitx-configtool at `/bin/fcitx-configtool`.

2.  Config GUI for gtk2:

    **Config GUI for gtk2 not found.**

3.  Config GUI for gtk3:

    Found `fcitx-config-gtk3` at `/bin/fcitx-config-gtk3`.

4.  Config GUI for kde:

    **`kcmshell4` not found.**

# Frontends setup:
## Xim:
1.  `${XMODIFIERS}`:

    Environment variable XMODIFIERS is set to "@im=fcitx" correctly.
    Xim Server Name from Environment variable is fcitx.

2.  XIM_SERVERS on root window:

    Xim server name is the same with that set in the environment variable.

## Qt:
1.  qt4 - `${QT4_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

2.  qt5 - `${QT_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

3.  Qt IM module files:

    Found fcitx qt module: `/usr/lib/fcitx/qt/libfcitx-quickphrase-editor5.so`.
    Found fcitx im module for Qt4: `/usr/lib/qt/plugins/inputmethods/qtim-fcitx.so`.
    Found fcitx im module for Qt5: `/usr/lib/qt5/plugins/platforminputcontexts/libfcitxplatforminputcontextplugin.so`.

## Gtk:
1.  gtk - `${GTK_IM_MODULE}`:

    Environment variable GTK_IM_MODULE is set to "fcitx" correctly.

2.  `gtk-query-immodules`:

    1.  gtk 2:

        Found `gtk-query-immodules` for gtk `2.24.32` at `/bin/gtk-query-immodules-2.0`.
        Version Line:

            # Created by /bin/gtk-query-immodules-2.0 from gtk+-2.24.32

        Found fcitx im modules for gtk `2.24.32`.

            "/usr/lib/gtk-2.0/2.10.0/immodules/im-fcitx.so" 
            "fcitx" "Fcitx (Flexible Input Method Framework)" "fcitx" "/usr/share/locale" "ja:ko:zh:*" 

    2.  gtk 3:

        Found `gtk-query-immodules` for gtk `3.24.8` at `/bin/gtk-query-immodules-3.0`.
        Version Line:

            # Created by /bin/gtk-query-immodules-3.0 from gtk+-3.24.8

        Found fcitx im modules for gtk `3.24.8`.

            "/usr/lib/gtk-3.0/3.0.0/immodules/im-fcitx.so" 
            "fcitx" "Fcitx (Flexible Input Method Framework)" "fcitx" "/usr/share/locale" "ja:ko:zh:*" 

3.  Gtk IM module cache:

    1.  gtk 2:

        Found immodules cache for gtk `2.24.32` at `/usr/lib/gtk-2.0/2.10.0/immodules.cache`.
        Version Line:

            # Created by usr/bin/gtk-query-immodules-2.0 from gtk+-2.24.32

        Found fcitx im modules for gtk `2.24.32`.

            "/usr/lib/gtk-2.0/2.10.0/immodules/im-fcitx.so" 
            "fcitx" "Fcitx (Flexible Input Method Framework)" "fcitx" "/usr/share/locale" "ja:ko:zh:*" 

    2.  gtk 3:

        Found immodules cache for gtk `3.24.8` at `/usr/lib/gtk-3.0/3.0.0/immodules.cache`.
        Version Line:

            # Created by usr/bin/gtk-query-immodules-3.0 from gtk+-3.24.8

        Found fcitx im modules for gtk `3.24.8`.

            "/usr/lib/gtk-3.0/3.0.0/immodules/im-fcitx.so" 
            "fcitx" "Fcitx (Flexible Input Method Framework)" "fcitx" "/usr/share/locale" "ja:ko:zh:*" 

4.  Gtk IM module files:

    1.  gtk 2:

        All found Gtk 2 immodule files exist.

    2.  gtk 3:

        All found Gtk 3 immodule files exist.

# Configuration:
## Fcitx Addons:
1.  Addon Config Dir:

    Found fcitx addon config directory: `/usr/share/fcitx/addon`.

2.  Addon List:

    1.  Found 27 enabled addons:

            fcitx-autoeng
            fcitx-chewing
            fcitx-chttrans
            fcitx-classic-ui
            fcitx-clipboard
            fcitx-dbus
            fcitx-freedesktop-notify
            fcitx-fullwidth-char
            fcitx-imselector
            fcitx-ipc
            fcitx-ipcportal
            fcitx-keyboard
            fcitx-kimpanel-ui
            fcitx-notificationitem
            fcitx-pinyin-enhance
            fcitx-pinyin
            fcitx-punc
            fcitx-quickphrase
            fcitx-remote
            fcitx-spell
            fcitx-table
            fcitx-unicode
            fcitx-vk
            fcitx-x11
            fcitx-xim
            fcitx-xkb
            fcitx-xkbdbus

    2.  Found 1 disabled addons:

            fcitx-qw

3.  Addon Libraries:

    All libraries for all addons are found.

4.  User Interface:

    Found 2 enabled user interface addons:

        fcitx-classic-ui
        fcitx-kimpanel-ui

## Input Methods:
1.  Found 2 enabled input methods:

        fcitx-keyboard-us-dvp
        chewing

2.  Default input methods:

    You have a keyboard input method "fcitx-keyboard-us-dvp" correctly added as your default input method.

# Log:
1.  `date`:

        Fri 12 Apr 2019 09:46:19 AM UTC

2.  `~/.config/fcitx/log/`:

        total 4
        -rw-r--r-- 1 username username 52 Apr 12 09:41 crash.log

3.  `~/.config/fcitx/log/crash.log`:

        fcitx: RenderBadPicture (invalid Picture parameter)

```

@CorporalCrawdad can you provide your fcitx-diagnose output? did you try to update your void, uninstall all of these fcitx packages, remove `$HOME/.config/fcitx/` and try fcitx-chewing again? does that still work for you?

CorporalCrawdad commented:
As requested. After discovering I had a misnamed .xinitrc, I can confirm that input switching (via ctrl+shift+space) works flawlessly. Worth noting, dwm does not have a tray, but it can be patched to have one.
`# System Info:
1.  `uname -a`:

        Linux venus 4.19.32_1 #1 SMP PREEMPT Wed Mar 27 20:41:38 UTC 2019 x86_64 GNU/Linux

2.  `lsb_release -a`:

        LSB Version:	1.0
        Distributor ID:	VoidLinux
        Description:	Void Linux
        Release:	rolling
        Codename:	void

3.  `lsb_release -d`:

        Description:	Void Linux

4.  `/etc/lsb-release`:

    `/etc/lsb-release` not found.

5.  `/etc/os-release`:

        NAME="void"
        ID="void"
        DISTRIB_ID="void"
        PRETTY_NAME="void"

6.  Desktop Environment:

    Cannot determine desktop environment.

7.  Bash Version:

        BASH_VERSION='5.0.3(1)-release'

# Environment:
1.  DISPLAY:

        DISPLAY=':1'

2.  Keyboard Layout:

    1.  `setxkbmap`:

            xkb_keymap {
            	xkb_keycodes  { include "evdev+aliases(qwerty)"	};
            	xkb_types     { include "complete"	};
            	xkb_compat    { include "complete"	};
            	xkb_symbols   { include "pc+us+us(dvorak):2+inet(evdev)"	};
            	xkb_geometry  { include "pc(pc105)"	};
            };

    2.  `xprop`:

            _XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us,us", ",dvorak", ""

3.  Locale:

    1.  All locale:

            C
            POSIX
            en_US.utf8

    2.  Current locale:

            LANG=en_US.UTF-8
            LC_CTYPE="en_US.UTF-8"
            LC_NUMERIC="en_US.UTF-8"
            LC_TIME="en_US.UTF-8"
            LC_COLLATE=C
            LC_MONETARY="en_US.UTF-8"
            LC_MESSAGES="en_US.UTF-8"
            LC_PAPER="en_US.UTF-8"
            LC_NAME="en_US.UTF-8"
            LC_ADDRESS="en_US.UTF-8"
            LC_TELEPHONE="en_US.UTF-8"
            LC_MEASUREMENT="en_US.UTF-8"
            LC_IDENTIFICATION="en_US.UTF-8"
            LC_ALL=

4.  Directories:

    1.  Home:

            /home/robert

    2.  `${XDG_CONFIG_HOME}`:

        Environment variable `XDG_CONFIG_HOME` is not set.

        Current value of `XDG_CONFIG_HOME` is `~/.config` (`/home/robert/.config`).

    3.  Fcitx Settings Directory:

        Current fcitx settings directory is `~/.config/fcitx` (`/home/robert/.config/fcitx`).

5.  Current user:

    The script is run as robert (1000).

# Fcitx State:
1.  executable:

    Found fcitx at `/bin/fcitx`.

2.  version:

    Fcitx version: `4.2.9.5`

3.  process:

    Found 2 fcitx processes:

        28114 fcitx
        28129 fcitx-dbus-watc

4.  `fcitx-remote`:

    `fcitx-remote` works properly.

5.  DBus interface:

    Using `dbus-send` to check dbus.

    Owner of DBus name `org.fcitx.Fcitx` is `:1.0`.

    PID of DBus name `org.fcitx.Fcitx` owner is `28114`.

# Fcitx Configure UI:
1.  Config Tool Wrapper:

    Found fcitx-configtool at `/bin/fcitx-configtool`.

2.  Config GUI for gtk2:

    **Config GUI for gtk2 not found.**

3.  Config GUI for gtk3:

    Found `fcitx-config-gtk3` at `/bin/fcitx-config-gtk3`.

4.  Config GUI for kde:

    **`kcmshell4` not found.**

# Frontends setup:
## Xim:
1.  `${XMODIFIERS}`:

    Environment variable XMODIFIERS is set to "@im=fcitx" correctly.
    Xim Server Name from Environment variable is fcitx.

2.  XIM_SERVERS on root window:

    Xim server name is the same with that set in the environment variable.

## Qt:
1.  qt4 - `${QT4_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

2.  qt5 - `${QT_IM_MODULE}`:

    Environment variable QT_IM_MODULE is set to "fcitx" correctly.

3.  Qt IM module files:

    **Cannot find fcitx input method module for Qt4.**
    **Cannot find fcitx input method module for Qt5.**

## Gtk:
1.  gtk - `${GTK_IM_MODULE}`:

    Environment variable GTK_IM_MODULE is set to "fcitx" correctly.

2.  `gtk-query-immodules`:

    1.  gtk 2:

        Found `gtk-query-immodules` for gtk `2.24.32` at `/bin/gtk-query-immodules-2.0`.
        Version Line:

            # Created by /bin/gtk-query-immodules-2.0 from gtk+-2.24.32

        **Failed to find fcitx in the output of `/bin/gtk-query-immodules-2.0`**

        **Cannot find fcitx im module for gtk 2.**

    2.  gtk 3:

        Found `gtk-query-immodules` for gtk `3.24.7` at `/bin/gtk-query-immodules-3.0`.
        Version Line:

            # Created by /bin/gtk-query-immodules-3.0 from gtk+-3.24.7

        **Failed to find fcitx in the output of `/bin/gtk-query-immodules-3.0`**

        **Cannot find fcitx im module for gtk 3.**

3.  Gtk IM module cache:

    1.  gtk 2:

        Found immodules cache for gtk `2.24.32` at `/usr/lib/gtk-2.0/2.10.0/immodules.cache`.
        Version Line:

            # Created by usr/bin/gtk-query-immodules-2.0 from gtk+-2.24.32

        **Failed to find fcitx in immodule cache at `/usr/lib/gtk-2.0/2.10.0/immodules.cache`**

        **Cannot find fcitx im module for gtk 2 in cache.**

    2.  gtk 3:

        Found immodules cache for gtk `3.24.7` at `/usr/lib/gtk-3.0/3.0.0/immodules.cache`.
        Version Line:

            # Created by usr/bin/gtk-query-immodules-3.0 from gtk+-3.24.7

        **Failed to find fcitx in immodule cache at `/usr/lib/gtk-3.0/3.0.0/immodules.cache`**

        **Cannot find fcitx im module for gtk 3 in cache.**

4.  Gtk IM module files:

    1.  gtk 2:

        All found Gtk 2 immodule files exist.

    2.  gtk 3:

        All found Gtk 3 immodule files exist.

# Configuration:
## Fcitx Addons:
1.  Addon Config Dir:

    Found fcitx addon config directory: `/usr/share/fcitx/addon`.

2.  Addon List:

    1.  Found 27 enabled addons:

            fcitx-autoeng
            fcitx-chewing
            fcitx-chttrans
            fcitx-classic-ui
            fcitx-clipboard
            fcitx-dbus
            fcitx-freedesktop-notify
            fcitx-fullwidth-char
            fcitx-imselector
            fcitx-ipc
            fcitx-ipcportal
            fcitx-keyboard
            fcitx-kimpanel-ui
            fcitx-notificationitem
            fcitx-pinyin-enhance
            fcitx-pinyin
            fcitx-punc
            fcitx-quickphrase
            fcitx-remote
            fcitx-spell
            fcitx-table
            fcitx-unicode
            fcitx-vk
            fcitx-x11
            fcitx-xim
            fcitx-xkb
            fcitx-xkbdbus

    2.  Found 1 disabled addons:

            fcitx-qw

3.  Addon Libraries:

    All libraries for all addons are found.

4.  User Interface:

    Found 2 enabled user interface addons:

        fcitx-classic-ui
        fcitx-kimpanel-ui

## Input Methods:
1.  Found 3 enabled input methods:

        chewing
        fcitx-keyboard-us
        fcitx-keyboard-us-dvorak

2.  Default input methods:

    **Your first (default) input method is chewing instead of a keyboard input method. You may have trouble deactivate fcitx.**

# Log:
1.  `date`:

        Sat 13 Apr 2019 01:27:07 AM CST

2.  `~/.config/fcitx/log/`:

        total 4
        -rw-r--r-- 1 robert wheel 631 Mar 26 20:50 crash.log

3.  `~/.config/fcitx/log/crash.log`:

        =========================
        FCITX 4.2.9.5 -- Get Signal No.: 11
        Date: try "date -d @1553604610" if you are using GNU date ***
        ProcessID: 874
        fcitx(+0x15b7)[0x5591de2e55b7]
        /usr/lib/libc.so.6(+0x3a410)[0x7f439acfa410]
        /usr/lib/libdbus-1.so.3(dbus_watch_get_enabled+0x5)[0x7f439a77f4d5]
        /usr/lib/fcitx/fcitx-dbus.so(+0x2326)[0x7f439a7aa326]
        /usr/lib/fcitx/fcitx-dbus.so(+0x2511)[0x7f439a7aa511]
        /usr/lib/libfcitx-core.so.0(+0x9d2c)[0x7f439b2e9d2c]
        /usr/lib/libfcitx-core.so.0(FcitxInstanceRun+0x330)[0x7f439b2ea7c0]
        fcitx(+0xefc)[0x5591de2e4efc]
        /usr/lib/libc.so.6(__libc_start_main+0xeb)[0x7f439ace6b6b]
        fcitx(+0xf7a)[0x5591de2e4f7a]

`

ixearth commented:
@CorporalCrawdad @maxice8 
I figured it out, it's problem of libchewing.
I found a post on fcitx google forum about this.
https://groups.google.com/d/msg/fcitx/ggYKSb-xFeA/bbJSNyDIAAAJ
currently libchewing only contains one file cuz it's build with cmake. we need to change to automake or have a separate package for data of libchewing.
having just one so file absolutely won't make libchewing functional.
@CorporalCrawdad chewing works for you probably because you build and installed libchewing with automake before. sudo make uninstall, uninstall fcitx-chewing and reinstall fcitx-chewing, libchewing will not works.

```
$ xbps-query -Rf libchewing
/usr/lib/libchewing.so.3.3.1
/usr/lib/libchewing.so.3 -> /usr/lib/libchewing.so.3.3.1
```

archlinux build libchewing with `./configure --prefix=/usr && make` (I have tried it on void-musl-x86_64 and it works)
https://www.archlinux.org/packages/community/x86_64/libchewing/ (check "Package Contents")
https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/libchewing
debian have a separate package for data of libchewing
https://packages.debian.org/stretch/libchewing3
https://packages.debian.org/stretch/libchewing3-data

mvf commented:
It looks like the libchewing data ended up in the `libchewing-devel` package due to an overzealous `vmove usr/share`.

maxice8 commented:
https://github.com/void-linux/void-packages/commit/8079269c76f7a135fcf01c11f31d40fd3105af4b

ixearth commented:
thanks. I can confirm fcitx-chewing works now (with libchewing-0.5.1_2).
